validation/src/main/scala/hmda/validation/ValidationStats.scala

```diff
@@ -22,7 +23,14 @@ object ValidationStats {
   case class ValidationStatsState(stats: Seq[SubmissionStats] = Nil) {
     def updated(event: Event): ValidationStatsState = event match {
       case SubmissionStatsAdded(s) =>
-        ValidationStatsState(stats :+ s)
+        val matchingSubs = stats.filter(stat => stat.id == s.id)
+        if (matchingSubs.isEmpty) {
+          ValidationStatsState(stats :+ s)
+        } else {
+          val oldSub = matchingSubs.head
+          val newSub = oldSub.copy(id = s.id, totalLars = oldSub.totalLars + s.totalLars, taxId = oldSub.taxId + s.taxId)
+          ValidationStatsState(stats.filterNot(s => s == oldSub) :+ newSub)
```

Comment:
This `s` is confusing because that's the name of a different variable earlier in this method. Could you change one of them to something clearer?